### ISSUE TITLE
Openseadragon Zoomify tile source with AWS S3 presigned URL 

### ISSUE THREAD

**Issue by dabomb1004** [openseadragon/openseadragon#1947]
Dear openseadragon community!

I am really excited to use openseadragon for our civil engineering inspection platform.  

I can successfully render images using openseadragon when the pyramids are stored in the ec2 but unable to do the same when the pyramids are stored in a private s3 server. I am trying to figure out how to call the pyramids using presigned url. I can get specific images to render. However, I am unable to render the pyramids.  

```js
AWS.config.accessKeyId = '******************';
AWS.config.secretAccessKey = '************************************';
AWS.config.region = "**********";

const s3 = new AWS.S3();
const s3_bucket_url_js = function($key) {
s3.getSignedUrl('getObject', {
			Bucket: "nwu-deliverables",
			Expires: 60*30,
			Key: $key,
	}, function (err, url) {
		if (err) {
		 console.log(err);
	 } else {
		 return url;
	 }
});
}

OpenSeadragon({
    id:            "openseadragon1",
    prefixUrl:     "/DTSS/assets/buttons",
		showNavigator: false,
    tileSources:   {
			  type: 'zoomifytileservice',
        height: 3600,
        width:  2700,
        tileSize: 256,
        // tilesUrl: 'assets/pyr/1310/'
				getTileUrl: function( level, x, y ){
            var url_s = "DTSS/assets/pyr/1310/SPAN01_VISUAL/TileGroup0/" +
                    level + "-" + x + "-" + y + ".jpg";
						return s3_bucket_url_js(url_s);
        }
    }
});


```

**Comment by SamHall** [comment]
Your using a different approach than I do - but I store my files on S3 and it works well.
I reference the .dzi file as a tilesource. 
To use it this way, I had to make the permissions 'public'. S3 warns against that, however, I think this is a valid use of read-only rights. 

Example:
tileSources: [
  "https://s3.amazonaws.com/dzchattzoom/whiteoak2.dzi",
   "https://s3.amazonaws.com/dzchattzoom/011_B1_DZ.dzi",
...
My website has several examples in use:  https://chattanoogahistory.com

**Comment by dabomb1004** [comment]
Sam sorry for the late response. We ended up modifying the openseadragon.min.js file we used and added an s3 presigner function in javascript to the line in around 5370 which has the getTileUrl for Zoomify ` getTileUrl: function(e, t, i) {
        var n;
        var o = this._calculateAbsoluteTileNumber(e, t, i);
        n = Math.floor(o / 256);
        var url = this.tilesUrl + "TileGroup" + n + "/" + e + "-" + t + "-" + i + ".jpg";
        return s3_bucket_url_js(url);


    }`

At the end above we added our s3_bucket_url_js function which uses the aws-sdk js version 2 in browser and return it. Our s3_bucket_url_js function is in the file where the div map is and uses `// AWS.config.region = 'us-east-1';
AWS.config.credentials = new AWS.CognitoIdentityCredentials({IdentityPoolId: 'us-east-1:1a588721-b00b-4155-9f74-1e5c6ad26de2' });
AWS.config.region = "us-east-1";



const s3 = new AWS.S3();
function s3_bucket_url_js_sync(key) {
  var params = {Bucket: 'nwu-deliverables', Key: key, Expires:1000};
  var s3_presigned = s3.getSignedUrl('getObject', params);
  return s3_presigned;
}
` 

We used AWS Cognito Identity Pool Id and allowed unauthenticated access and attached an IAM policy for s3 read only access to the role attached to the Identity Pool allowing us to not use our secret access key in the browser.
